docs/user-guides/dashboard-as-code.md

```diff
@@ -0,0 +1,89 @@
+# Dashboard-as-Code
+
+Perses offers the possibility to define dashboards as code instead of going through manipulations on the UI.
+But why would you want to do this? Basically Dashboard-as-Code (DaC) is something that becomes useful
+at scale, when you have many dashboards to maintain, to keep aligned on certain parts, etc.
+
+DaC benefits can be summarized as follows:
+- **Operational efficiency**, by the reusability of a common basis of dashboard components for standard monitoring use cases.
+- **Implementation cost reduction**, by leveraging on existing components when creating new dashboards.
+- **Maintenance cost reduction**, by making it easier to: cascade the same update on multiple dashboards, keep multiple components aligned with each other, etc..
+
+Most of these benefits comes from not dealing with the Perses JSON format directly: instead, we work with [CUE](https://cuelang.org/), a powerful templating language with a strong emphasis on data validation, that also brings the factorization & dependency management capabilities (well, [not yet!](https://github.com/cue-lang/cue/discussions/2330)) we need here.
+
+Also, as-code means it's GitOps-friendly, meaning that you can also benefit from:
+- versions history
+- peer-review of changes before rollout
+- automated deployments
+- and more..
+
+## Getting started
+
+### Prerequisites
+
+- `percli`, the [CLI of Perses](../tooling/cli.md).
+- `cue`, the [CLI of Cuelang](https://cuelang.org/).
+- [Go](https://go.dev/).
```

Comment:
that's a weird things to require `Go` when you are providing a `cue` lib, don't you think ?
I don't even really get why do we need Go here

Reply:
Needed for dep management (you need to `go get` before being able to `cue get go`) https://github.com/perses/perses/pull/1664/files#diff-6f12614d0527f3d0486999783e4b2656892df91b600ae4017eae65c58a7829d9R48

Hopefully this is something that will vanish when CUE will release its dep management?

Reply:
Go prereq removed since we now commit the generated file (DaC users can retrieve these ones directly)